secret-manager/runtime/src/main/java/io/quarkiverse/googlecloudservices/secretmanager/runtime/config/SecretManagerConfigSource.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkiverse.googlecloudservices.secretmanager.runtime.config;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.google.cloud.secretmanager.v1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import io.smallrye.config.common.AbstractConfigSource;
+
+public class SecretManagerConfigSource extends AbstractConfigSource {
+
+    /** The ordinal is set to < 100 (which is the default) so that this config source is retrieved from last. */
+    private static final int SECRET_MANAGER_ORDINAL = 50;
```

Comment:
Yes, I think it would be preferable to be last -- 

This SecretManagerConfigSource will get values for properties in the form: `sm//<SECRET_ID>/<VERSION>`. Any other normal property that comes through (like in the form `quarkus.some.property`) will return null. 

The below code for `getValue(..)` shows what happens where we will parse the property name that comes in, and if it doesn't match the form `sm//....` we just return null.

```
    @Override
    public String getValue(String propertyName) {
        SecretVersionName secretVersionName = SecretManagerConfigUtils.getSecretVersionName(propertyName, defaultProject);
        if (secretVersionName == null) {
            // The propertyName is not in the form "${sm//...}" so return null.
            return null;
        }

        AccessSecretVersionResponse response = clientProvider.get().accessSecretVersion(secretVersionName);
        return response.getPayload().getData().toStringUtf8();
    }
```

So it wouldn't be too bad to be last. And the purpose of this is to allow users to be able to reference secrets from their `application.properties` file like:

```
quarkus.database.password=${sm//my-db-password}
```